scripts/openstack-update-services.sh

```diff
@@ -0,0 +1,160 @@
+fi
+if [ -z "$NAMESPACE" ]; then
+    echo "Please set NAMESPACE"; exit 1
+fi
+
+BASE_DIR="${UPDATE_ARTIFACT_DIR}"
+if [ -e "${BASE_DIR}" ]; then
+    BASE_DIR="${BASE_DIR}/"
+else
+    BASE_DIR=""                 # current directory
+fi
+
+# The update_event.sh is created ci-framework update role.
+update_event() {
+    local event="${@:-Unknown Event}"
+    local event_script="${UPDATE_ARTIFACT_DIR}/update_event.sh"
+    if [ -e "${event_script}" ]; then
+        "${event_script}" "${event}"
+    fi
+}
+
+OPENSTACK_OPERATOR_CSV=$(oc get csv -n $OPERATOR_NAMESPACE -o name | grep openstack-operator)
+OPENSTACK_VERSION_CR=$(oc get openstackversion -n $NAMESPACE -o name)
+
+if [ "${FAKE_UPDATE}" != "false" ]; then
+    oc get $OPENSTACK_OPERATOR_CSV -o yaml -n $OPERATOR_NAMESPACE  > $OUTFILE
+    sed -i $OUTFILE -e "s|value: .*/$CONTAINERS_NAMESPACE/\(.*\)@:.*|value: quay.io/$CONTAINERS_NAMESPACE/\1:$CONTAINERS_TARGET_TAG|g"
+    OPENSTACK_DEPLOYED_VERSION=$(oc get -n $NAMESPACE $OPENSTACK_VERSION_CR --template={{.status.deployedVersion}})
+    sed -i $OUTFILE -e "s|value: $OPENSTACK_DEPLOYED_VERSION|value: $OPENSTACK_VERSION|"
+
+    update_event Applying Fake Update CR
+
+    oc apply -f $OUTFILE
+fi
+
+oc project $NAMESPACE
+# wait until openstackVersion cr completes reconcile, status.availableVersion should be the same as VERSION
+oc wait $OPENSTACK_VERSION_CR --for=jsonpath='{.status.availableVersion}'=$OPENSTACK_VERSION --timeout=$TIMEOUT
+
+
+OPENSTACK_DEPLOYED_VERSION=$(oc get $OPENSTACK_VERSION_CR --template={{.status.deployedVersion}})
+
+cat <<EOF >openstackversionpatch.yaml
+    "spec": {
+      "targetVersion": "$OPENSTACK_VERSION"
+      }
+EOF
+
+update_event Patching the Openstack Version
+
+oc patch $OPENSTACK_VERSION_CR  --type=merge  --patch-file openstackversionpatch.yaml
+
+# wait for ovn update on control plane
+oc wait $OPENSTACK_VERSION_CR --for=condition=MinorUpdateOVNControlplane --timeout=$TIMEOUT
+
+update_event MinorUpdateOVNControlplane Completed
+
+# start ovn update on data plane
+nodes_with_ovn=()
+# Get the names of all OpenStackDataPlaneNodeSet resources
+openstackdataplanenodesets=$(oc get openstackdataplanenodeset -o custom-columns=NAME:.metadata.name,SERVICES:.spec.services --no-headers)
+
+# Loop through each OpenStackDataPlaneNodeSet
+while read -r node_name services; do
+    # Check if 'ovn' is in the list of services
+    for service in ${services[@]};do
+        if [[ "$service" == *"ovn"* ]]; then
+            nodes_with_ovn+=("- $node_name")
+            break
+        fi
+    done
+done <<< $openstackdataplanenodesets
+
+DATAPLANE_DEPLOYMENT=edpm
+OVN_NODE_SETS=$(printf '    %s\n' "${nodes_with_ovn[@]}")
+
+cat <<EOF >edpm-deployment-ovn-update.yaml
+apiVersion: dataplane.openstack.org/v1beta1
+kind: OpenStackDataPlaneDeployment
+metadata:
+  name: $DATAPLANE_DEPLOYMENT-ovn-update
+spec:
+  nodeSets:
+$OVN_NODE_SETS
+  servicesOverride:
+    - ovn
+EOF
+
+update_event Applying the OVN CRD
```

Comment:
should this be `Applying the EDPM OVN update deployment CR` ? its not applying an CRD

Reply:
Fixed.